File: flutter_app/lib/models/department.dart
class Department {
  final String id;
  final String name;
  final String code;
  final String description;
  final String building;
  final String floor;
  final String contactEmail;
  final String contactPhone;

  Department({
    required this.id,
    required this.name,
    required this.code,
    required this.description,
    required this.building,
    required this.floor,
    required this.contactEmail,
    required this.contactPhone,
  });

  factory Department.fromJson(Map<String, dynamic> json) {
    return Department(
      id: json['_id'] ?? '',
      name: json['name'] ?? '',
      code: json['code'] ?? '',
      description: json['description'] ?? '',
      building: json['building'] ?? '',
      floor: json['floor'] ?? '',
      contactEmail: json['contactEmail'] ?? '',
      contactPhone: json['contactPhone'] ?? '',
    );
  }
}

File: backend/models/Department.js
const mongoose = require('mongoose');

const departmentSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
    unique: true,
    trim: true
  },
  code: {
    type: String,
    required: true,
    unique: true,
    uppercase: true,
    trim: true
  },
  hodId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Faculty',
    default: null
  },
  description: {
    type: String,
    default: ''
  },
  building: {
    type: String,
    default: ''
  },
  floor: {
    type: String,
    default: ''
  },
  contactEmail: {
    type: String,
    default: ''
  },
  contactPhone: {
    type: String,
    default: ''
  },
  establishedYear: {
    type: Number,
    default: null
  },
  totalFaculty: {
    type: Number,
    default: 0
  },
  totalStudents: {
    type: Number,
    default: 0
  },
  isActive: {
    type: Boolean,
    default: true
  }
}, {
  timestamps: true
});

// Index for faster queries
departmentSchema.index({ name: 1, code: 1 });

module.exports = mongoose.model('Department', departmentSchema);
